src/ui/osx/TogglDesktop/Feature/Timeline/TimelineDatasource.swift

```diff
@@ -286,18 +300,107 @@ extension TimelineDatasource: TimelineTimeEntryCellDelegate {
 
 extension TimelineDatasource: TimelineBaseCellDelegate {
 
-    func timelineCellMouseDidExited(_ sender: TimelineBaseCell) {}
-
     func timelineCellMouseDidEntered(_ sender: TimelineBaseCell) {
         switch sender {
         case let timeEntryCell as TimelineTimeEntryCell:
             guard let timeEntry = timeEntryCell.timeEntry else { return }
             delegate?.shouldPresentTimeEntryHover(in: timeEntryCell.popoverView, timeEntry: timeEntry)
-        case let activityCell as TimelineActivityCell:
-            guard let activity = activityCell.activity else { return }
-            delegate?.shouldPresentActivityHover(in: sender.view, activity: activity)
         default:
             break
         }
     }
+
+    func timelineCellUpdateEndTime(with event: NSEvent, sender: TimelineBaseCell) {
+        isUserResizing = false
+        switch sender {
+        case let timeEntryCell as TimelineTimeEntryCell:
+            guard let timeEntry = timeEntryCell.timeEntry else { return }
+            let endAt = convertPointForEndTime(with: event, startTime: timeEntry.start)
+
+            // Update in libray
+            delegate?.shouldUpdateEndTime(endAt, for: timeEntry)
+        default:
+            break
+        }
+    }
+
+    func timelineCellRedrawEndTime(with event: NSEvent, sender: TimelineBaseCell) {
+        isUserResizing = true
+        switch sender {
+        case let timeEntryCell as TimelineTimeEntryCell:
+            guard let timeEntry = timeEntryCell.timeEntry else { return }
+            let endAt = convertPointForEndTime(with: event, startTime: timeEntry.start)
+
+            // Update the end time and re-draw
+            timeEntry.end = endAt
+            flow.invalidateLayout()
```

Comment:
In order to prevent updating the Time during the Resize Session, we don't update the data in Library, we just re-draw the TimeEntry.

When the session is done -> Update to the library

Reply:
This is a very good way to do it. No need to save it extensively as the status will change. Only at the end of the resize it makes sense to save the object.